libkbfs/folder_branch_ops.go

```diff
@@ -1014,20 +1005,20 @@ func (fbo *folderBranchOps) nowUnixNano() int64 {
 }
 
 func (fbo *folderBranchOps) maybeUnembedAndPutOneBlock(ctx context.Context,
-	md *RootMetadata) error {
+	md *RootMetadata) (*blockPutState, error) {
 	if fbo.config.BlockSplitter().ShouldEmbedBlockChanges(&md.data.Changes) {
-		return nil
+		return nil, nil
 	}
 
 	_, uid, err := fbo.config.KBPKI().GetCurrentUserInfo(ctx)
 	if err != nil {
-		return err
+		return nil, err
 	}
 
 	bps := newBlockPutState(1)
 	err = fbo.unembedBlockChanges(ctx, bps, md, &md.data.Changes, uid)
 	if err != nil {
-		return err
+		return nil, err
```

Comment:
nit: seems cleaner to have `unembedBlockChanges` call `newBlockPutState` and return the `bps`